modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -381,6 +387,64 @@ public void unShareB2BApplication() throws JSONException {
     }
 
     @Test(dependsOnMethods = "unShareB2BApplication")
+    public void testOnboardChildOrganization() throws IOException {
+
+        String body = readResource("add-smaller-hospital-organization-request-body.json");
+        body = body.replace("${parentId}", organizationID);
+        HttpPost request = new HttpPost(serverURL + TENANT_PATH + tenant + PATH_SEPARATOR + ORGANIZATION_PATH
+                                    + API_SERVER_PATH + ORGANIZATION_MANAGEMENT_API_BASE_PATH);
+        Header[] headerList = new Header[3];
+        headerList[0] = new BasicHeader("Authorization", "Bearer " + switchedM2MToken);
+        headerList[1] = new BasicHeader(CONTENT_TYPE_ATTRIBUTE, "application/json");
+        headerList[2] = new BasicHeader(HttpHeaders.ACCEPT, "application/json");
+        request.setHeaders(headerList);
+        request.setEntity(new StringEntity(body));
+        HttpResponse response = client.execute(request);
+        Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_CREATED);
+
+        String jsonResponse = EntityUtils.toString(response.getEntity());
+        JsonObject responseObject = JsonParser.parseString(jsonResponse).getAsJsonObject();
+        childOrganizationID = responseObject.get("id").getAsString();
+        assertNotNull(childOrganizationID);
+    }
+
+    @DataProvider(name = "dataProviderForGetOrganizationsMetaAttributes")
+    public Object[][] dataProviderForGetOrganizationsMetaAttributes() {
+
+        return new Object[][] {
+                {"?filter=attributes eq Country&limit=10&recursive=false", false, false},
+                {"?filter=attributes eq Region&limit=10&recursive=true", true, false},
+                {"?filter=attributes sw C and attributes ew try&limit=10&recursive=false", false, false},
+                {"?filter=attributes co A&limit=10&recursive=true", false, true},
+        };
+    }
+
+    @Test(dependsOnMethods  = "testOnboardChildOrganization",
+          dataProvider      = "dataProviderForGetOrganizationsMetaAttributes")
+    public void testGetOrganizationsMetaAttributes(String filterQuery, boolean isRecursive, boolean expectEmptyList) {
```

Comment:
other than the filter query, limit and recursive query params have been passed to the first
Can we fix the data provider?

Reply:
This is now updated to only pass the filter as the first param.